### ISSUE TITLE
Missing seekbackward and seekforward in PlayerProviderContext

### ISSUE THREAD

**Issue by danielr18** [benwiley4000/cassette#305]
I'm sure this could be achieved with a combination of `onSeekPreview` + timeout `onSeekComplete` (due to having to wait for state to change), but I think it would be easier to have functions for `seekbackward` and `seekforward` in the context, similar to the MediaSession ones.

`seekbackward: (seekLength) => (this.media.currentTime -= seekLength)`;
`seekforward: (seekLength) => (this.media.currentTime += seekLength)`;

**Comment by benwiley4000** [comment]
Right, this makes sense. The onSeekPreview + onSeekComplete weren't really designed to do this.

However I wonder if the best thing might be to allow `onSeekComplete` to take a parameter (the seek time) so you can use it without `onSeekPreview` to skip immediately. You could do:

```jsx
seekForward() {
  const { onSeekComplete, currentTime } = playerContext;
  onSeekComplete(currentTime + 5);
}
seekBackward() {
  const { onSeekComplete, currentTime } = playerContext;
  onSeekComplete(currentTime - 5);
}
```

Seems pretty straightforward and more flexible than adding extra API for that one use case. What do you think?

**Comment by benwiley4000** [comment]
I made some typos the first time I wrote that comment so make sure to re-read. :) sorry.

**Comment by danielr18** [comment]
I like it, that's how I tried to use `onSeekComplete ` in the beginning until I went to the source code and saw how it worked.

**Comment by benwiley4000** [comment]
@danielr18 that sounds like a good sign that this is how it *should* work then! 😄 

If you want to make a PR and update some documentation, that would be great - otherwise I can try to get to it later.

**Comment by danielr18** [comment]
I can make the PR :)

**Comment by danielr18** [comment]
I have a weird bug that isn't really related to seek, but the ProgressDisplay component. 
![image](https://user-images.githubusercontent.com/9046616/48921418-6bff8d00-ee7e-11e8-8371-a07f76e20ca9.png)

Somehow when I click ~2 times over the seek buttons, red squares in the image, the player seeks to the position above of where I clicked. If you want, you can try to recreate it here: http://54.224.96.152:3001/, perhaps it's a bug we can solve in the library?

**Comment by benwiley4000** [comment]
@danielr18 that's a really peculiar bug! I'm able to repro it as well when I visit your site. Without access to the source code it might be hard for me to debug it. Is it possible to either share the code, or if not, share a minimal repro that demonstrates the bug so I can mess around with it locally?

**Comment by danielr18** [comment]
I'll try to create a minimal repro, and share it

**Comment by benwiley4000** [comment]
Thanks so much! And once you do, if you could log this as a separate issue that would be much appreciated.

**Comment by danielr18** [comment]
Yes, planning to do that, we can close this issue.

**Comment by benwiley4000** [comment]
Closed by #309.
